### DISCUSSION TITLE
Dynamic vacuum cards

### DISCUSSION THREAD

**Discussion by szamfirov** [Clooos/Bubble-Card#926] [Feature requests]
Hey @Clooos,

I'm a huge fan of bubble-card and have been using it for quite a while - thank you for the massive amount of work that you've put in!

Some context before I ask my question - I have a single page dashboard with pop-ups which represent the rooms in my house. I'm using auto-entities for a lot of the content in those pop-ups to dynamically list room-specific entities as bubble-cards.

I'm struggling to make the vacuum cards dynamic because of the subbutton actions (i.e `call-service`) which need a `target_entity` - those cannot be passed to the card as a template. Is there a way to implicitly pick up which entity id to use as a target for things like `vacuum.start` / `vacuum.stop` etc.?

That would massively help with any dynamically created cards or just templating in general.

Cheers,
Svet

**Comment by Clooos** [comment]
Hi! I just changed that in my code. In the next beta if there is no entity defined in the sub-button actions, it will automatically take the sub-button entity for the action, and if there is no entity in the sub-button it will automatically take the card entity.

This is so much better like that indeed, just tried on my vacuum card and this works perfectly!

**Comment by szamfirov** [comment]
That sounds awesome, looking forward to it!

**Reply by Clooos** [reply]
The beta is here, tell me if everything works as it should 😄

https://github.com/Clooos/Bubble-Card/releases/tag/v2.3.0-beta.7

**Reply by szamfirov** [reply]
Works great! Just had to replace:
```yaml
target:
  entity_id: vacuum.dusty
```
with:
```yaml
target: {}
```
in all the sub-buttons 👌 

Apologies for the delay, I went down the rabbit hole of also giving the climate entity card a try and was changing things in my dashboard with all that goodness 😅

**Reply by Clooos** [reply]
I have a problem with this approach 😅 It breaks services that don't use any entity, I will have to change how it works again, I will try something like that in the next beta:

```yaml
target:
  entity_id: entity
```

And 

```yaml
target:
  entity_id: sub_button_entity
```

**Reply by Clooos** [reply]
Here is the new release, take a look at the changelog 🙂

https://github.com/Clooos/Bubble-Card/releases/tag/v2.3.0-beta.8

**Reply by szamfirov** [reply]
Just tested it and it seems to works absolutely fine by using `entity_id: entity`. Thanks for the updates! 🙏 

PS: also the climate card is working flawlessly so far - I barely have to go to the climate details pop-up because everything is visible on the button itself 👌

**Reply by Clooos** [reply]
I'm so glad to read that! 😄
